app/experimenter/experiments/api/v6/serializers.py

```diff
@@ -41,13 +41,15 @@ def get_feature(self, obj):
         feature_value = {}
         feature_enabled = False
 
-        if obj.experiment.feature_configs.exists():
-            feature_config = obj.experiment.feature_configs.all().order_by("slug").first()
+        feature_configs = obj.experiment.feature_configs.all()
+        feature_values = obj.feature_values.all()
+        if feature_configs:
+            feature_config = sorted(feature_configs, key=lambda f: f.slug)[0]
             feature_config_slug = feature_config.slug
-            if obj.feature_values.filter(feature_config=feature_config).exists():
-                branch_feature_value = obj.feature_values.get(
-                    feature_config=feature_config
-                )
+            if feature_config in [fv.feature_config for fv in feature_values]:
+                branch_feature_value = [
+                    fv for fv in feature_values if fv.feature_config == feature_config
+                ][0]
```

Comment:
As much as I hate that this has to have so much name repetition, you've made this really readable 😂 👏